pkg/rebuild/pypi/infer.go

```diff
@@ -41,40 +41,69 @@ import (
 
 // These are commonly used in PyPi metadata to point to the project git repo, using a map as a set.
 // Some people capitalize these differently, or add/remove spaces. We normalized to lower, no space.
-var commonRepoLinks = map[string]bool{
-	"homepage":     true,
-	"home":         true,
-	"sourcecode":   true,
-	"source":       true,
-	"repository":   true,
-	"github":       true,
-	"project":      true,
-	"issuetracker": true,
+// This list is ordered, we will choose the first occurance.
+var commonRepoLinks = []string{
+	"source",
+	"sourcecode",
+	"repository",
+	"project",
+	"github",
 }
 
+// There are two places to find the repo:
+// 1. In the ProjectURLs (project links)
+// 2. Embeded in the description
+//
+// For 1, there are some ProjectURLs that are very common to use for a repo
+// (commonRepoLinks above), so we can break up the ProjectURLs
+
+// Preference:
+// where               | known repo
+// --------------------------------
+// project source link | yes
+// project source link | no
+// description         | yes
+// other project links | yes
+
 func (Rebuilder) InferRepo(t rebuild.Target, mux rebuild.RegistryMux) (string, error) {
 	project, err := mux.PyPI.Project(t.Package)
 	if err != nil {
 		return "", nil
 	}
+	var repoLinks []string
 	for name, url := range project.ProjectURLs {
-		_, ok := commonRepoLinks[strings.ReplaceAll(strings.ToLower(name), " ", "")]
-		if !ok {
-			continue
+		for _, commonName := range commonRepoLinks {
+			if strings.ToLower(name) == commonName {
+				repoLinks = append(repoLinks, url)
+				break
+			}
 		}
-		// Exclude sponsor URLs.
+	}
+	// Four priority levels:
+	// 1. project source link | known repo
```

Comment:
nit: the pipe here isn't making this super clear. maybe a comma e.g. "project source link, known repo" or "AND"?